src/components/ec-table/ec-table.vue

```diff
@@ -71,80 +71,80 @@
   </div>
 </template>
 
-<script setup>
+<script setup lang="ts">
 import { useMediaQuery } from '@vueuse/core';
 import { computed, useAttrs, useSlots } from 'vue';
 
 import EcTableFooter from '../ec-table-footer';
 import EcTableHead from '../ec-table-head';
+import type { StickyColumnPosition, TableHeadColumn, TableHeadSort } from '../ec-table-head/types';
+import type { TableEvent, TableEvents } from './types';
 
 const isInCustomRowThreshold = useMediaQuery('(max-width: 768px)');
 const slots = useSlots();
 const attrs = useAttrs();
 // const emit = defineEmits(['sort', 'row-click']);
-const emit = defineEmits(['sort']);
-
-const props = defineProps({
-  columns: {
-    type: Array,
-    default: () => [],
-  },
-  sorts: {
-    type: Array,
-    default: () => [],
-  },
-  data: {
-    type: Array,
-    default: () => [],
-  },
-  totalRecords: {
-    type: Number,
-  },
-  maxHeight: String,
-  stickyColumn: {
-    type: String,
-    validator(value) {
-      return ['left', 'right'].includes(value);
-    },
-  },
-  title: String,
-  isCustomRowShown: {
-    type: Boolean,
-    default: () => undefined,
-  },
-  isTableHeaderHidden: {
-    type: Boolean,
-    default: () => undefined,
-  },
+const emit = defineEmits<{
+  'sort': [value: TableEvents[TableEvent.SORT]],
+}>();
+
+interface TableProps {
+  columns?: TableHeadColumn[],
+  sorts?: TableHeadSort[],
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: any[],
+  totalRecords?: number,
+  maxHeight?: string,
+  stickyColumn?: StickyColumnPosition,
+  title?: string,
+  isCustomRowShown?: boolean,
+  isTableHeaderHidden?: boolean
+}
+
+const props = withDefaults(defineProps<TableProps>(), {
+  columns: () => [],
+  sorts: () => [],
+  data: () => [],
+  isCustomRowShown: undefined,
+  isTableHeaderHidden: undefined,
```

Comment:
```suggestion
```

Reply:
Those two default values can't be removed, otherwise they will be set to `false` by default instead of `undefined`

Reply:
https://play.vuejs.org/#eNp9U01v2zAM/SucLnaAzGlR7JLaGbYuwDYMW9EG2MUXxaYTdbJkSHISwPB/HyU3bpBluYnk4+Pjhzr2qWmSXYtszlJbGNE4sOjaZpErUTfaOOjAYDWFPXfFdllVWDjooTK6hojyohH3oOvm1Z/MvOFpKZyrQivroLYbyDxXHH1FKTX81kaW76LJESDsD81LoUaYHEyPyNVJ/TieQLYgYWNGsuOyRcqjIsP7HjyplphIvYmjk+xo0hMhQK6o0ZWoUbee0ROO2cQUWV1jdEZDAEqfwoebm5vJ/QWGfxVd5BF27I3Ybokt0KWzYQU0fDIc1o3kDskCSLe3i64LQ+z7dEZW8ArVtA5272tdosxyRvGcDSG/gkUaNkGOdDbSsSlzluRUYpO8WK1o9V0YCCsIKySaX40TJDdncwgRH+O0sv334HOmxenRX2yx+HPB/2IP3pezR4MWzQ5zNsYcNxt0Q3j5/BMP9B6D1EorCX0l+IQ0ytZrHGCfW1WS7BNcUPstHCZNeWWXB4fKHpvyQj2yD/ic0aH6Mf2v9Te5d8ldyMtVT1M8Hvn53wHJ1YaW4YjG/yPl0FS8QFjxtcRHoxs7kAv70Fqn6ye9f97qvfo4h7WmI+GKhFGR4V80ISGDvXDbL1jxVjobl1gJNXClb7SLeDK9TD0HmlHIKUPT/g9cOTe6n1LsXp/CBg1QcWlx3nVBUHJeAbIsGyB0oGsTSE+zx/o02WsUI+5IE9TMBjmnZ9z/Bd1jnNc= those should be set to undefined if we don't pass anything as their values

Reply:
Probably I missed something but... that's what I already got isn't it? I mean I already set those to undefined inside `withDefaults`

Reply:
You could remove it, as that default assignment of undegfined doesn't make sense :)